### DISCUSSION TITLE
undefined output from print

### DISCUSSION THREAD

**Discussion by agardnerIT** [pyodide/pyodide#4358] [Q&A]
Hi all, just found pyodide and it looks fantastic!

I believe I'm trying to do the most simple thing and yet getting an `undefined` output. Any ideas why?

```
let prog = 'print("hello world")';
let output = pyodide.runPython(prog);
addToOutput(output);
```

**Comment by ryanking13** [comment]
`pyodide.runPython` returns a value of the last expression. Since `print` does not return anything (it just prints the output to stdout), the return is `undefined`.

If you want to redirect the stdout to a variable, you can use [`pyodide.setStdout`](https://pyodide.org/en/stable/usage/streams.html):

```js
const arr = [];
pyodide.setStdout({ batched: (msg) => arr.push(msg) });
pyodide.runPython("print('ABC')");
console.log(arr)
```
